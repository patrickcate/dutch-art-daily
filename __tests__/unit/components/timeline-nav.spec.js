import TimelineNav from '@components/timeline-nav.vue'

describe('TimelineNav Component', () => {
  let wrapper
  let options

  beforeEach(() => {
    wrapper = createWrapper(TimelineNav, options, {
      state: {
        slides: [],
      },
      actions: {
        updatePaginationNumber() {},
      },
    })

    wrapper.vm.$root.swipers.timeline = {
      params: {
        slidesPerView: null,
      },
    }
  })

  it('should be a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('beforeMount hook is called', () => {
    const beforeMountMock = jest.fn()

    wrapper = createWrapper(TimelineNav, options, {
      state: {
        slides: [],
      },
      actions: {
        updatePaginationNumber() {
          beforeMountMock()
        },
      },
    })

    expect(beforeMountMock).toHaveBeenCalled()
  })

  // TODO: Test that pagination number changes based on screen size.
})
