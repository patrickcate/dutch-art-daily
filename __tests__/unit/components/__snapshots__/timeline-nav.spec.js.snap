// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineNav Component renders correctly 1`] = `
<div class="timeline-nav">
  <base-carousel-stub name="timeline" slide-number="3" class="timeline-wrapper"></base-carousel-stub>
</div>
`;
